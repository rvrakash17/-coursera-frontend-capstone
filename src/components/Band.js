/**
 * Component for showing details of the Band.
 *
 * @component
 * @example
 * const band = {
 *         name:bandName,
 *         festivals:[
 *            {
 *               name:"LOL-palooza"
 *            }
 *         ]
 *   };
 *
 * return (
 *   <li>Winter Primates<ul><li>LOL-palooza</li></ul></li>
 * )
 */
export default function Band(props) {
  return (
    <li>
      <span data-testid="band-band-name">{props.band.name}</span>
      <ul>
        {props.band.festivals &&
          props.band.festivals.map(function (festival, index) {
            return (
              <li data-testid="band-festival-name" key={index}>
                {festival.name ? festival.name : ""}
              </li>
            );
          })}
      </ul>
    </li>
  );
}
