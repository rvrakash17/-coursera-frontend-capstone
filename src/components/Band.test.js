import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Band from "./Band";

describe("Band Component", () => {
  // Test 1
  test("Band Rendering with name and festivals", () => {
    /* Fixtures */
    const bandName = "Winter Primates";
    const band = {
      name: bandName,
      festivals: [
        {
          name: "LOL-palooza",
        },
        {
          name: "MIDI",
        },
      ],
    };
    render(<Band band={band} key={0} />);
    const band_band_name = screen.getByTestId("band-band-name");
    expect(band_band_name).toBeInTheDocument();

    // snapshot
    const domTree = renderer.create(<Band band={band} key={0} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  // Test 2
  test("Band Rendering with empty festivals", () => {
    /* Fixtures */
    const bandName = "Winter Primates";
    const band = {
      name: bandName,
      festivals: [],
    };
    render(<Band band={band} key={0} />);
    const band_band_name = screen.getByTestId("band-band-name");
    expect(band_band_name).toBeInTheDocument();
    // snapshot
    const domTree = renderer.create(<Band band={band} key={0} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  // Test 3
  test("Band Rendering with missing festivals", () => {
    /* Fixtures */
    const bandName = "Winter Primates";
    const band = {
      name: bandName,
    };
    render(<Band band={band} key={0} />);
    const band_band_name = screen.getByTestId("band-band-name");
    expect(band_band_name).toBeInTheDocument();

    // snapshot
    const domTree = renderer.create(<Band band={band} key={0} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
