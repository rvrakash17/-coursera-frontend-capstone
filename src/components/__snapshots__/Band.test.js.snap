// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Band Component Band Rendering with empty festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul />
</li>
`;

exports[`Band Component Band Rendering with missing festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul />
</li>
`;

exports[`Band Component Band Rendering with name and festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul>
    <li
      data-testid="band-festival-name"
    >
      LOL-palooza
    </li>
    <li
      data-testid="band-festival-name"
    >
      MIDI
    </li>
  </ul>
</li>
`;
